extensions/razer/driver/razerkbd_driver.c

```diff
@@ -0,0 +1,1576 @@
+{
+    struct razer_report report = razer_chroma_standard_set_led_effect(VARSTORE, BACKLIGHT_LED, 0x02);
+    razer_send_payload(usb_dev, &report);
+
+    UInt16 product = -1;
+    (*usb_dev)->GetDeviceProduct(usb_dev, &product);
+
+    switch (product)
+    {
+    case USB_DEVICE_ID_RAZER_BLACKWIDOW_STEALTH:
+    case USB_DEVICE_ID_RAZER_BLACKWIDOW_STEALTH_EDITION:
+    case USB_DEVICE_ID_RAZER_BLACKWIDOW_ULTIMATE_2012:
+    case USB_DEVICE_ID_RAZER_BLACKWIDOW_ULTIMATE_2013:
+        report = razer_chroma_standard_set_led_effect(VARSTORE, LOGO_LED, 0x02);
+        break;
+    }
+
+    return count;
+}
+
+/**
+ * Read device file "mode_pulsate"
+ *
+ * Returns a string
+ */
+ssize_t razer_attr_read_mode_pulsate(IOUSBDeviceInterface **usb_dev, char *buf)
+{
+    struct razer_report report = razer_chroma_standard_get_led_effect(VARSTORE, LOGO_LED);
+    struct razer_report response = razer_send_payload(usb_dev, &report);
+
+    return sprintf(buf, "%d\n", response.arguments[2]);
+}
+
+/**
+ * Read device file "profile_led_red"
+ *
+ * Actually a Yellow LED
+ *
+ * Returns a string
+ */
+ssize_t razer_attr_read_tartarus_profile_led_red(IOUSBDeviceInterface **usb_dev, char *buf)
+{
+    struct razer_report report = razer_chroma_standard_get_led_state(VARSTORE, RED_PROFILE_LED);
+    struct razer_report response = razer_send_payload(usb_dev, &report);
+
+    // CHRIS MODIFIED:
+    //return sprintf(buf, "%d\n", response.arguments[2]);
+    
+    return response.arguments[2];
+}
+
+/**
+ * Read device file "profile_led_green"
+ *
+ * Returns a string
+ */
+ssize_t razer_attr_read_tartarus_profile_led_green(IOUSBDeviceInterface **usb_dev, char *buf)
+{
+    struct razer_report report = razer_chroma_standard_get_led_state(VARSTORE, GREEN_PROFILE_LED);
+    struct razer_report response = razer_send_payload(usb_dev, &report);
+
+    // CHRIS MODIFIED:
+    //return sprintf(buf, "%d\n", response.arguments[2]);
+    
+    return response.arguments[2];
+    
+}
+
+/**
+ * Read device file "profile_led_blue"
+ *
+ * Returns a string
+ */
+ssize_t razer_attr_read_tartarus_profile_led_blue(IOUSBDeviceInterface **usb_dev, char *buf)
+{
+    struct razer_report report = razer_chroma_standard_get_led_state(VARSTORE, BLUE_PROFILE_LED);
+    struct razer_report response = razer_send_payload(usb_dev, &report);
+
+    // CHRIS MODIFIED:
+    //return sprintf(buf, "%d\n", response.arguments[2]);
+    
+    return response.arguments[2];
+}
+
+/**
+ * Read device file "get_firmware_version"
+ *
+ * Returns a string
+ */
+ssize_t razer_attr_read_get_firmware_version(IOUSBDeviceInterface **usb_dev, char *buf)
+{
+    struct razer_report report = razer_chroma_standard_get_firmware_version();
+    struct razer_report response_report = razer_send_payload(usb_dev, &report);
+    
+    // MODIFIED BY CHRIS:
+    //return sprintf(buf, "v%d.%d\n", response_report.arguments[0], response_report.arguments[1]);
```

Comment:
@asmagill - This code was crashing. Any suggestions on how I can "join" these two unsigned char's?

Reply:
Ignore me... I implemented a (lazy) workaround.